src/main/frontend/components/EditReview.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router";
+import _ from "lodash";
+import ErrorList from "./ErrorList";
+import VenueField from "./VenueField";
+
+const EditReview = (props) => {
+  const [formPayload, setFormPayload] = useState({
+    eventName: "",
+    userName: "",
+    text: "",
+    rating: "",
+    venueId: "",
+  });
+
+  const [errors, setErrors] = useState({});
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const id = props.id;
+  const venueId = props.venueId;
+
+  const fetchReview = async () => {
+    try {
+      const response = await fetch(`/api/v1/reviews/${id}`);
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      const responseBody = await response.json();
+      console.log(responseBody);
+      setFormPayload(responseBody.review);
+    } catch (error) {
+      console.error(`There was an error in fetch: ${error}`);
+    }
+  };
+
+  const editReview = async () => {
+    try {
+      const response = await fetch(`/api/v1/reviews/${id}`, {
+        method: "PUT",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(formPayload),
+      });
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json();
+          return setErrors(body.errors);
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      }
+      const body = await response.json();
+      setShouldRedirect(true);
+    } catch (err) {
+      console.error(`Error in fetch: ${err.message}`);
+    }
+  };
+
+  const validForSubmission = () => {
+    const errors = {};
+    for (const field in formPayload) {
+      if (
+        field !== "text" &&
+        field !== "venueId" &&
+        field !== "rating" &&
+        field !== "venue" &&
+        field !== "id" &&
+        formPayload[field].trim() === ""
+      ) {
+        errors[field] = "is blank";
+      }
+    }
+    setErrors(errors);
+    return _.isEmpty(errors);
+  };
+
+  const resetForm = () => {
+    setFormPayload({
+      eventName: "",
+      userName: "",
+      text: "",
+      rating: "",
+      venueId: "",
+    });
+    setErrors({});
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (validForSubmission()) {
+      editReview(formPayload);
+      resetForm();
```

Comment:
Since you will be redirecting on a successful submission, you can remove the `resetForm` functionality here